### PR TITLE
docs(cas-backends): reference docs

### PR DESCRIPTION
**Pull request by migmartri**
## Reference section

image

## Getting started

Removed all the content from setup, I left the page because I believe we link it from the repo

image

Updated the bottom section of the workflow-creation view to mention cas-backends as optional step

image

I also updated many links in the docs to point to the new CAS backend doc

refs https://github.com/chainloop-dev/chainloop/issues/223

### CODE REVIEW

File: docs/faq.md [hunk 1]
```diff
@@ -24,4 +24,4 @@ That said, there are [benefits](/reference/operator/contract#runner-context) for
 
 #### Does Chainloop store my Artifacts and Attestation metadata?
 
-No. They are stored in [your OCI Registry](/getting-started/setup#add-oci-repository).
+No. They are stored in [your CAS backend](/reference/operator/cas-backend).
```
Comment (danlishka) [inline docs/faq.md:27]: your Content-Addressed Storage (CAS)

File: docs/getting-started/workflow-definition.mdx [hunk 2]
```diff
@@ -176,6 +176,18 @@ Save the following token since it will not printed again:
...
 We have everything we need to integrate our CI with Chainloop!
 
-## Third-Party integrations
+## Optional setup
+
+### CAS Backend
+
+As part of an attestation process, you might want to collect different pieces of evidence such as Software Bill Of Materials (SBOMs), test results, runner logs, etc and then attach them to the final in-toto attestation. 
+
+By default, Chainloop comes pre-configured with what we call an `inline` backend. The inline backend **embeds** the pieces of evidence in the resulting attestations. This is useful to get started quickly but since the pieces of evidences are embedded in the attestation, their size is limited.
```
Comment (danlishka) [inline docs/getting-started/workflow-definition.mdx:185]: pieces of evidences -> pieces of evidence
We have three "pieces of evidence" in a row. Should we use SSC metadata, just metadata or materials?

File: docs/reference/operator/cas-backend/cas-backend.mdx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+---
+
+import Image from "@theme/IdealImage";
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+import CodeBlock from "@theme/CodeBlock";
+
+As part of an attestation process, you might want to collect different pieces of evidence such as Software Bill Of Materials (SBOMs), test results, runner logs, etc and then attach them to the final in-toto attestation. 
+
+Chainloop helps with this process by providing a Content Addressable Storage API proxy that
```
Comment (danlishka) [inline docs/reference/operator/cas-backend/cas-backend.mdx:13]: that -> that: ?

File: docs/reference/operator/cas-backend/cas-backend.mdx [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+:::info
+New CAS Backends will be added over time. If yours is not implemented yet, please [let us know](https://chainloop.dev/contact)
+:::
+
+### Inline (fallback)
+
+Chainloop comes pre-configured with what we call an `inline` backend.
+
+The inline backend **embeds** the pieces of evidence in the resulting attestations. This is useful to get started quickly but since the pieces of evidences are embedded in the attestation, their size is limited.
```
Comment (danlishka) [inline docs/reference/operator/cas-backend/cas-backend.mdx:52]: pieces of evidences -> p of evidence

File: docs/reference/operator/cas-backend/cas-backend.mdx [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+
+## Manage backends
+
+You can setup as many CAS backends as you want, but you can only have **one enabled as default at the time**.  This **default backend will be used** during the attestation process **to store the pieces of evidence**.
+
+In Chainloop, CAS backends can be managed with the `chainloop cas-backend` command.
+
+
+```bash title="List backends"
```
Comment (danlishka) [inline docs/reference/operator/cas-backend/cas-backend.mdx:30]: Listing backends

File: docs/reference/operator/cas-backend/cas-backend.mdx [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+import Image from "@theme/IdealImage";
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+import CodeBlock from "@theme/CodeBlock";
+
+As part of an attestation process, you might want to collect different pieces of evidence such as Software Bill Of Materials (SBOMs), test results, runner logs, etc and then attach them to the final in-toto attestation. 
+
+Chainloop helps with this process by providing a Content Addressable Storage API proxy that 
+
+- **Abstracts away the underlying storage backend**. Currently, we support OCI registries as storage backend but you can expect blob storage, Artifactory and other storage backends to be supported in the future.
```
Comment (danlishka) [inline docs/reference/operator/cas-backend/cas-backend.mdx:15]: as storage backends

**Review by danlishka — APPROVED** [review]
LGTM. Please take a look at my comments.

File: docs/getting-started/workflow-definition.mdx (hunk 2 above)
Comment (migmartri) [reply docs/getting-started/workflow-definition.mdx:185]: I like it, thanks

**Pull request merged**